sale_delivery_rate/models/delivery_carrier_rate.py

```diff
@@ -25,9 +29,9 @@ class DeliveryCarrierRate(models.Model):
     @api.multi
     def generate_equiv_picking_rates(self, stock_picking):
         stock_picking.ensure_one()
-        recordset_data = self.copy_data()
-
-        for record_data in recordset_data:
-            del record_data['sale_order_id']
-            record_data['picking_id'] = stock_picking.id
-            self.env['stock.picking.rate'].create(record_data)
+        for record in self:
+            recordset_data = record.copy_data()
+            for record_data in recordset_data:
```

Comment:
There's no need for an inner loop here now that you're calling `copy_data` against a single record. The result will be a single dictionary inside a list, so you can just do `record_data = record.copy_data()[0]` instead.